### PR TITLE
mediathekview: Add desktop file and add JVM options

### PR DESCRIPTION
**Pull request by avdv**
## Description of changes

1. add a desktop file, so that mediathekview is available from launchers

2. add the recommended JVM options to the wrapper. These options are checked for on startup and a warning dialog is displayed each time if these flags are not passed (the check can be disabled using the `-nj, --no-jvm-param-checks` CLI option).

3. avoid an `UnsatisfiedLinkError` for libprism_es2.so on Linux, the library depends on the xf86vm xorg extension

Fixes #288255

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/mediathekview/default.nix [hunk 1]
```diff
@@ -37,6 +38,18 @@ stdenv.mkDerivation rec {
     runHook postInstall
   '';
 
+  desktopItems = makeDesktopItem {
+    name = "MediathekView";
+    exec = meta.mainProgram;
+    icon = "MediathekView";
+    desktopName = meta.mainProgram;
+    genericName = "MediathekView";
+    comment = meta.description;
```
Comment (SuperSandro2000) [inline pkgs/applications/video/mediathekview/default.nix:58]: A change to meta should ideally not trigger a rebuild

File: pkgs/applications/video/mediathekview/default.nix [hunk 2]
```diff
@@ -14,6 +14,16 @@ stdenv.mkDerivation rec {
   installPhase =
   let
     libraryPath = lib.strings.makeLibraryPath [ libglvnd libnotify ];
+    jvmArgList = [
+      "-XX:+UseShenandoahGC"
+      "-XX:ShenandoahGCHeuristics=compact"
+      "-XX:MaxRAMPercentage=50.0"
+      "-XX:+UseStringDeduplication"
+      "--add-opens=java.desktop/sun.awt.X11=ALL-UNNAMED"
+      "-Dfile.encoding=UTF-8"
```
Comment (SuperSandro2000) [inline pkgs/applications/video/mediathekview/default.nix:23]: Can we link the source of them?

File: pkgs/applications/video/mediathekview/default.nix [hunk 3]
```diff
@@ -13,7 +13,7 @@ stdenv.mkDerivation rec {
 
   installPhase =
   let
-    libraryPath = lib.strings.makeLibraryPath [ libglvnd libnotify ];
+    libraryPath = lib.strings.makeLibraryPath ([ libglvnd libnotify ] ++ lib.optional stdenv.isLinux xorg.libXxf86vm);
```
Comment (SuperSandro2000) [inline pkgs/applications/video/mediathekview/default.nix:16]:
```suggestion
    libraryPath = lib.makeLibraryPath ([ libglvnd libnotify ]
      ++ lib.optional stdenv.isLinux xorg.libXxf86vm);
```

File: pkgs/applications/video/mediathekview/default.nix (hunk 2 above)
Comment (avdv) [reply pkgs/applications/video/mediathekview/default.nix:25]: Done.

File: pkgs/applications/video/mediathekview/default.nix (hunk 1 above)
Comment (avdv) [reply pkgs/applications/video/mediathekview/default.nix]: Good call. I just set those attributes directly now, without referring to `meta`.

**Comment by avdv** [comment]
Thanks for the review @SuperSandro2000 :heart: PTAL

**Comment by avdv** [comment]
@SuperSandro2000 is there anything left to do?

File: pkgs/applications/video/mediathekview/default.nix [hunk 4]
```diff
@@ -8,35 +8,61 @@ stdenv.mkDerivation rec {
...
     install -m644 MediathekView.jar $out/lib
+    install -m644 MediathekView.svg $out/share/pixmaps
 
     makeWrapper ${jre}/bin/java $out/bin/mediathek \
-      --add-flags "-jar $out/lib/MediathekView.jar" \
+      --add-flags "${jvmArgs} -jar $out/lib/MediathekView.jar" \
       --suffix LD_LIBRARY_PATH : "${libraryPath}"
 
     makeWrapper ${jre}/bin/java $out/bin/MediathekView \
       --add-flags "-jar $out/lib/MediathekView.jar" \
```
Comment (Moredread) [inline pkgs/applications/video/mediathekview/default.nix:43]: This line needs to be removed. I get an error when runnung MediathekView:

```
./results/mediathekview/bin/MediathekView
Picked up _JAVA_OPTIONS: -Dsun.java2d.uiScale=2
Unknown options: '-XX:+UseShenandoahGC', '-XX:ShenandoahGCHeuristics=compact', '-XX:MaxRAMPercentage=50.0', '-XX:+UseStringDeduplication', '--add-opens=java.desktop/sun.awt.X11=ALL-UNNAMED', '-Dfile.encoding=UTF-8', '-DexternalUpdateCheck', '-jar'
```

Comment (Moredread) [reply pkgs/applications/video/mediathekview/default.nix:43]: Besides that LGTM, but I can't test the desktop items part, as I'm using a tiling wm.

Comment (avdv) [reply pkgs/applications/video/mediathekview/default.nix]: > This line needs to be removed. I get an error when runnung MediathekView:

Nice catch! Fixed. BTW, do you know why there are two scripts that start the app? I just tested `mediathek`...

> Besides that LGTM, but I can't test the desktop items part, as I'm using a tiling wm.

I use xmonad myself. And rofi. It shows up there.

Comment (Moredread) [reply pkgs/applications/video/mediathekview/default.nix]: > > This line needs to be removed. I get an error when runnung MediathekView:
> 
> Nice catch! Fixed. BTW, do you know why there are two scripts that start the app? I just tested `mediathek`...

Those where the scripts in the binary release version for Linux in previous versions. I haven't looked at that archive for some time, but it looks like things have changed in https://download.mediathekview.de/stabil/MediathekView-14.0.0-linux.tar.gz

There are now two launcher scripts (like on windows), ```MediathekView``` and ```MediathekView_Portable```, seemingly generated by install4j.

I guess we could either generate our own scripts with those names and VM options, try to patch the upstream scripts, or (finally) build the package ourselves.

The easiest option, generating our own wrappers like before, would be ok for now I think. Maybe we don't even need the portable version, though it seems to be included in the Deb package.

We should add it to the release note though that the older wrappers have been removed.

**Comment by Moredread** [comment]
> > This line needs to be removed. I get an error when runnung MediathekView:
> 
> Nice catch! Fixed. BTW, do you know why there are two scripts that start the app? I just tested `mediathek`...

Those where the scripts in the binary release version for Linux in previous versions. I haven't looked at that archive for some time, but it looks like things have changed in https://download.mediathekview.de/stabil/MediathekView-14.0.0-linux.tar.gz

There are now two launcher scripts (like on windows), ```MediathekView``` and ```MediathekView_Portable```, seemingly generated by install4j.

I guess we could either generate our own scripts with those names and VM options, try to patch the upstream scripts, or (finally) build the package ourselves.

The easiest option, generating our own wrappers like before, would be ok for now I think. Maybe we don't even need the portable version, though it seems to be included in the Deb package.

We should add it to the release note though that the older wrappers have been removed.

**Comment by Moredread** [comment]
After looking into it more, I'd say we should just remove the mediathek and MediathekView_ipv4 wrappers and keep our wrapper for MediathekView.

MediathekView_Portable just adds "Einstellungen/.mediathek3" as the first arg, so I'd say we can skip it.

**Comment by avdv** [comment]
Oh, time really is fleeting... Sory, I forgot abou this.

> After looking into it more, I'd say we should just remove the mediathek and MediathekView_ipv4 wrappers and keep our wrapper for MediathekView.
> 
> MediathekView_Portable just adds "Einstellungen/.mediathek3" as the first arg, so I'd say we can skip it.

Done. I also added  an entry to the release notes.

> I'd very much appreciate it if this gets merged in time to make it into the upcoming 24.05 stable release,

Well, that's too late now, but I can do a backport once this is merged.

**Comment by avdv** [comment]
> I didn't look to closely and was thinking that you wrote that one, so didn't check this PR anymore. I hope your work wasn't in vain.

Oh, not a big deal. It just confused me at first why my PR would have conflicts after rebasing, then I realized the other PR hasn't been merged either... 

Duplicate of https://github.com/NixOS/nixpkgs/pull/306292

**Pull request closed without merging**